### ISSUE TITLE
Why is the associated Connection type for RedisConnectionManager an Option?

### ISSUE THREAD

**Issue by seunlanlege** [khuey/bb8#66]
```rust

#[async_trait]
impl bb8::ManageConnection for RedisConnectionManager {
    type Connection = Option<Connection>;
    type Error = RedisError;

    async fn connect(&self) -> Result<Self::Connection, Self::Error> {
        self.client.get_async_connection().await.map(Some)
    }

    async fn is_valid(&self, mut conn: Self::Connection) -> Result<Self::Connection, Self::Error> {
        // The connection should only be None after a failure.
        redis::cmd("PING")
            .query_async(conn.as_mut().unwrap())
            .await
            .map(|()| conn)
    }

    fn has_broken(&self, conn: &mut Self::Connection) -> bool {
        conn.is_none()
    }
}
```

the `is_valid` method also unwraps this option, is that safe? under what conditions is it safe?

**Comment by khuey** [comment]
See the original PR: https://github.com/khuey/bb8/pull/11

The redis APIs consume the connection on failure so we need something that we can move the connection object in and out of. See what happens in `RedisConnectionManager::run`. If an error is raised the connection (now `None`) will be returned to `put_back` which will call `has_broken` on it which will return true and the connection will be dropped. A consumer of this library should never see a Connection that is a `None`.

**Comment by djc** [comment]
Since 0.14.0, `redis::Cmd::query_async()` actually takes a`&mut redis::aio::Connection`, so I think we no longer need the `Option` wrapper here. Please have a look at #69 and see if it works for your use case.

**Comment by seunlanlege** [comment]
closed by #69
